[Hai Shan] iP

underthehai opened the pull request.

Commit:
Add Gradle support

oceancrust committed:
skeleton project

oceancrust committed:
level 1: greet, echo, exit

oceancrust committed:
added Task class

oceancrust committed:
Revert "added Task class"

This reverts commit 77b3425e01675d164870e10e094fbfbf09c6ac8c.

oceancrust committed:
level 2: Add, List items

oceancrust committed:
Level 3: Mark as Done

underthehai committed:
task string override

Signed-off-by: limhaishan <haishanlim@gmail.com>

underthehai committed:
introduce different types of tasks

underthehai committed:
level 4: ToDos, Event, Deadline updated

underthehai committed:
included exceptions

underthehai committed:
improve layout of divider
include exceptions catching invalid input and task with empty description
simplified main method

underthehai committed:
change data type from array to arraylist

underthehai committed:
Level 6: handle deletion of task

underthehai committed:
modified tester

underthehai committed:
round 1 testing successful

underthehai committed:
A-TextUiTesting: Automated Text UI Testing
set up properly with test input and expected output to compare

underthehai committed:
handle more errors from invalid input

underthehai committed:
Add file reading properties

Tasks from previous sessions are not recorded
Causes extra trouble for user to type in old information whenever the session restarts
Hence we can load data when Emily program starts by retrieving task infromation from the data folder

underthehai committed:
Save data to hard drive

Program could only read text file but not overwrite the file with new data

Included a new method to write over the text file and another method to
process the arraylist of task information to a string of text format

underthehai committed:
Update the understanding of date and time

Any input of string was considered a valid timestamp for the tasks

The task should accept valid timestamp in the form of yyyy-mm-dd

Update the timestamp for deadline and event to localdate type. Emily should handle exceptions thrown due to
timestamp

underthehai committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

underthehai committed:
Add type identity

underthehai committed:
Merge branch 'branch-Level-7'

underthehai committed:
Correct saving of tasks

Past commits neglected the cases of marking tasks as done and deletion of task

Information within text is not updated correctly

New changes account for the different commands

underthehai committed:
fix bugs

underthehai committed:
A-MoreOOP

underthehai committed:
Organise into packages

underthehai committed:
Add JUnit tests

Test toString of deadline and check that new task are assignened as unfinished

underthehai committed:
Add JavaDoc comments

Add header comments to methods and classes

underthehai committed:
Tweak the code to comply with a coding standard

underthehai committed:
Give users a way to find a task by searching for a keyword

underthehai committed:
Merge branch 'branch-A-JavaDoc'

underthehai committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/emily/command/Parser.java
#	src/main/java/emily/command/Storage.java
#	src/main/java/emily/command/Ui.java
#	src/main/java/emily/task/Deadline.java
#	src/main/java/emily/task/Event.java
#	src/main/java/emily/task/Task.java

underthehai committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/emily/command/Ui.java

underthehai committed:
Debug

underthehai committed:
Merge remote-tracking branch 'origin/add-gradle-support'

timjkong commented on src/main/java/emily/command/Parser.java:
```diff
@@ -0,0 +1,57 @@
+                return item;
+
+            }
+        }catch (ArrayIndexOutOfBoundsException e) {
```
Perhaps you could add a space between the curly bracket and catch.

timjkong commented on src/main/java/emily/command/Storage.java:
```diff
@@ -0,0 +1,114 @@
+                            break;
+
+                    }
+                }
```
Should indentation for case clauses be removed?

timjkong commented on src/main/java/emily/command/TaskList.java:
```diff
@@ -0,0 +1,41 @@
+
+    }
+
+    public ArrayList<Task> finder(String keyword){
```
I think there should be a space between the method and the curly bracket.

timjkong commented on src/main/java/emily/command/TaskList.java:
```diff
@@ -0,0 +1,41 @@
+
+    public ArrayList<Task> finder(String keyword){
+        ArrayList<Task> ls = new ArrayList<>();
+        for(Task t : this.store){
```
I think there should be a space between 'for' and the bracket and another space at the end before the curly bracket.

timjkong commented on src/main/java/emily/command/TaskList.java:
```diff
@@ -0,0 +1,41 @@
+        ArrayList<Task> ls = new ArrayList<>();
+        for(Task t : this.store){
+            String d = t.getDescription();
+            if(d.contains(keyword)){
```
I think there should be a space between 'if' and the bracket and another space at the end before the curly bracket.

underthehai replied:
Okay, thank you for pointing out!

timjkong commented on src/main/java/emily/command/Ui.java:
```diff
@@ -0,0 +1,107 @@
+        } else if (input.contains("todo") || input.contains("deadline") || input.contains("event")) {
+
+            if (input.trim().equals("todo") || input.trim().equals("deadline") || input.trim().equals("event")
+                    || input.trim().equals("deadline/by") || input.trim().equals("event/at")) {
```
I like how the line break comes before the || operator. The 8 space indentation is also good.

timjkong commented on src/main/java/emily/command/Ui.java:
```diff
@@ -0,0 +1,107 @@
+                current = p.process(input);
+                ls.add(current);
+                System.out.println("    Now you have " + (ls.retrieve().size()) + " tasks in the list" +
+                        "\n" + divider);
```
Perhaps the line break should come before the + operator.

timjkong commented on src/main/java/emily/command/Ui.java:
```diff
@@ -0,0 +1,107 @@
+            }
+
+        } else if (input.contains("done") || input.contains("delete") ||input.contains("find")) {
+            if(input.contains("done")){
```
I think there should be a space between 'if' and the bracket and another space at the end before the curly bracket.

timjkong commented on src/main/java/emily/command/Ui.java:
```diff
@@ -0,0 +1,107 @@
+                System.out.println(divider + "\n    Nice work, I have marked this task as done: " + current
+                        + "\n" + divider);
+
+            } else if(input.contains("delete")){
```
I think there should be a space between 'if' and the bracket and another space at the end before the curly bracket.

timjkong commented on src/main/java/emily/command/Ui.java:
```diff
@@ -0,0 +1,107 @@
+
+            } else if(input.contains("delete")){
+
+                if(input.trim().equals("delete")){
```
I think there should be a space between 'if' and the bracket and another space at the end before the curly bracket.

timjkong commented on src/main/java/emily/command/Ui.java:
```diff
@@ -0,0 +1,107 @@
+                System.out.println("    You have " + ls.retrieve().size() + " tasks in your list now" +
+                        "\n" + divider);
+            } else{ //case of finding
+                if(input.trim().equals("find")){
```
I think there should be a space between 'if' and the bracket and another space at the end before the curly bracket.

timjkong commented on src/main/java/emily/command/Ui.java:
```diff
@@ -0,0 +1,107 @@
+
+                System.out.println(divider + "\n    Here are the matching tasks in your list");
+                int index = 1;
+                for(Task t: arr){
```
I think there should be a space between 'for' and the bracket and another space at the end before the curly bracket.

timjkong commented on src/main/java/emily/task/Event.java:
```diff
@@ -0,0 +1,37 @@
+ */
+public class Event extends Task{
+
+public class Event extends Task {
```
I think you might have accidentally declared the Event class twice?

timjkong commented on src/main/java/emily/task/Event.java:
```diff
@@ -0,0 +1,37 @@
+    @Override
+    public String toString() {
+        return "[E]" + super.toString()
+                + "(at: " + this.at.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
I like how the line break comes before the + operator.

timjkong commented on src/main/java/emily/task/Task.java:
```diff
@@ -0,0 +1,39 @@
+        this.isFinished = b;
+    }
+
+    public boolean isFinished() {
```
The name of this boolean method sounds like a boolean, which is good.

timjkong commented on src/main/java/emily/command/Emily.java:
```diff
@@ -0,0 +1,64 @@
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    static String FILE_PATH = "data/emily.txt";
```
Perhaps the String should be 'private static final'? Either that, or it should not be capitalised and separated by underscores since it is not a constant.

underthehai replied:
Okay, thanks for pointing out! I forget to add in the private access modifier part

timjkong commented on src/main/java/emily/command/Emily.java:
```diff
@@ -0,0 +1,64 @@
+        boolean end = false;
+
+        System.out.println("Hello, I am Emily\n" +
+                "What can i do for you?\n" +
```
Perhaps the line breaks should come before the + operators.

timjkong reviewed (commented):
Overall, I think your code was very readable besides a few issues with indentation, spacing and naming of constants.

underthehai committed:
Adds checkstyle to gradle

underthehai committed:
Adds GUI functions

Current code creates a basic chat program based on java fx tutorial

Further customisation needs to be done for Emily program

dextertanyj commented on src/main/java/emily/command/Emily.java:
```diff
@@ -0,0 +1,68 @@
+
+    /**
+     * Initialises Emily bot
+     * @param filePath location to store the information
```
I think there should be an empty line between the description and parameter sections. I noticed the same issue in a few other places as well.

dextertanyj commented on src/main/java/emily/command/Emily.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    /**
+     * Interact with the user
```
I think we should use the word "Interacts" instead of "Interact"? I noticed the same issue in a few other places as well.

dextertanyj commented on src/main/java/emily/command/Storage.java:
```diff
@@ -0,0 +1,121 @@
+                    String[] temp;
+
+                    switch (type) {
+                        case 'T':
```
I think the indentation level of `case` should be inline with that of the `switch` statement?

underthehai replied:
Okay, I have corrected the indentations. Thanks for pointing them out!

dextertanyj commented on src/main/java/emily/task/Task.java:
```diff
@@ -0,0 +1,44 @@
+        this.isFinished = b;
+    }
+
+    public boolean isFinished() {
```
Perhaps `hasFinished` would be a better name for this method?

dextertanyj commented on src/main/java/emily/task/ToDos.java:
```diff
@@ -0,0 +1,22 @@
+    /**
+     * Subclass of Task with a given type
+     *
+     * @param description
```
I think there is a missing description for this parameter?

dextertanyj commented on src/test/java/emily/task/TaskTest.java:
```diff
@@ -0,0 +1,14 @@
+import main.java.emily.task.Task;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
I think it would be good to avoid wildcard imports? I noticed this in a few other places as well.

dextertanyj commented on src/main/java/emily/command/Ui.java:
```diff
@@ -0,0 +1,109 @@
+        Parser p = new Parser();
+        String divider = "    ---------------";
+
+        if (input.isEmpty() || input.trim().isEmpty()) {
```
I like the check for the trimmed version of the input!

dextertanyj reviewed (commented):
Overall, good job on following the coding standards. Besides the few minor touchups, perhaps the use of line breaks could be slightly more consistent? Other than that, it looks good to me!

underthehai committed:
Change program to GUI format

Users are able to launch the program and interact

underthehai committed:
Index error

Retrieve the wrong character as task index

underthehai committed:
Ends the application with user input

underthehai committed:
Merge branch 'branch-Level-10'

underthehai committed:
Update the codestyle based on the reviews provided

underthehai committed:
Add test for event timestamp

Timestamp is updated to include both date and time for event tasks

underthehai committed:
Generate new jar file

underthehai committed:
Update main.java as module

underthehai committed:
Add assertions and reorganise files

underthehai committed:
Merge pull request #1 from underthehai/branch-A-Assertions

Branch a assertions

underthehai committed:
Reorganise files

underthehai committed:
Merge branch 'master' of https://github.com/underthehai/ip

* 'master' of https://github.com/underthehai/ip:
  Add assertions and reorganise files
  Update main.java as module

# Conflicts:
#	src/main/java/emily/command/Ui.java
#	src/main/java/emily/storage/Storage.java
#	src/main/java/emily/storage/TaskList.java

underthehai committed:
Reorganise file

underthehai committed:
Examines the code and refactor to improve the code quality

underthehai committed:
Merge pull request #2 from underthehai/branch-A-CodeQuality

Examines the code and refactor to improve the code quality

underthehai committed:
Change output lines from emily into a stream

underthehai committed:
Set up Continuous Integration (CI)

underthehai committed:
Add extention to view task on a specific date

underthehai committed:
Add GUI features

underthehai committed:
Add feature to edit task description

underthehai committed:
Add readme

underthehai committed:
Set theme jekyll-theme-cayman

underthehai committed:
Add a representative screenshot

underthehai committed:
Merge branch 'master' of https://github.com/underthehai/ip

underthehai committed:
Add checks for correct input for edit and view commands

underthehai committed:
Push user guide commit

underthehai committed:
Update UserGuide in README.md

underthehai committed:
Fix index issues in Logic.java

underthehai committed:
Update tests

underthehai committed:
Update checkstyle
